### ISSUE TITLE
JavadocVariable doesn't report missing comment when excludeScope=anoninner

### ISSUE THREAD

**Issue by edrandall** [checkstyle/checkstyle#3183]
JavadocVariable - https://checkstyle.org/config_javadoc.html#JavadocVariable
 
does not report missing comments on member variables in a standard class,  irrespective of scope, when excludeScope="anoninner". 

config.xml:

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" 
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="ASCII" />
    <property name="severity" value="error" />
    <property name="fileExtensions" value="java" />

    <module name="TreeWalker">
        <module name="JavadocVariable">
            <property name="excludeScope" value="anoninner" />
        </module>
    </module>
</module>
```

TestClass.java:

```
package test;

/** Class for testing checkstyle. */
public class TestClass {

    private String key;

    public TestClass() {
    }
}
```

`$ java -jar checkstyle-6.18-all.jar -c config.xml TestClass.java`
Actual output:

```
Starting audit...
Audit done.
```

Expected output:

```
Starting audit...
[ERROR] C:\checkstyle\TestClass.java:6:9: Missing a Javadoc comment [JavadocVariable]
Audit done.
Checkstyle ends with 1 errors.
```
##

**Comment by rnveach** [comment]
Most likely this is because of issues we have had with `Scope` enumeration and should switch to `AccessModifier` here like we did for `ParameterNameCheck`.

**Comment by leojahn1** [comment]
I don't understand this problem. Based on this definition of Scope and excludeScope (https://checkstyle.sourceforge.io/property_types.html#Scope),
Scope is functioning properly. What are we trying to resolve here? Also, what reason would there be to set excludeScope=anoninner in the first place?

**Comment by cowwoc** [comment]
Per https://github.com/checkstyle/checkstyle/issues/9280 this issue should have been closed.

**Comment by romani** [comment]
Yes, no `excludeScope` after 10.22.0
